grafana_wtf/compat.py

```diff
@@ -0,0 +1,12 @@
+from sys import modules
+
+import niquests
+import urllib3
+
+# Amalgamate the module namespace to make all modules aiming
+# to use `requests`, in fact use `niquests` instead.
+modules["requests"] = niquests
+modules["requests.adapters"] = niquests.adapters
+modules["requests.sessions"] = niquests.sessions
+modules["requests.exceptions"] = niquests.exceptions
+modules["requests.packages.urllib3"] = urllib3
```

Comment:
@Ousret: Could this monkeypatching / amalgamation code be also provided by Niquests itself? To be invoked, for example, like:
```python
import niquests
niquests.patch_all()
```

Reply:
Unfortunately, this isn't the way to go. We should not tamper with sys.module outside of a development environment.

Try this instead

```python
import requests_cache
import niquests


class CacheSession(requests_cache.session.CacheMixin, niquests.Session):
    ...


if __name__ == "__main__":

    s = CacheSession()

    for i in range(60):
        r = s.get('https://httpbin.org/delay/1')

```

While its still not ideal, it should work more accordingly.
I am thinking about forking popular extensions and see how we can ease this.

Reply:
Thank you very much for your advise. We will submit a patch to work with requests-cache in the way you are suggesting.

Reply:
Converged that into a patch, thanks again.

- GH-135